### PR TITLE
premake: new, 5.0.0~beta1

### PR DESCRIPTION
**Pull request by xyq1113723547**
Topic Description
-----------------
premake: new, 5.0 alpha 16

Package(s) Affected
-------------------

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

File: extra-utils/premake/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# build script copied from Arch Linux
+abinfo "Building premake ..."
+cd "build/gmake2.unix"
+make STRIP="ture" config=release
```
Comment (KexyBiscuit) [inline extra-utils/premake/autobuild/build]: `ture`?

File: extra-utils/premake/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+abinfo "Installing premake and LISENSE"
```
Comment (KexyBiscuit) [inline extra-utils/premake/autobuild/build]: `LISENSE`?

File: extra-utils/premake/autobuild/build [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+cd ../..
+install -Dm755 "bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
```
Comment (KexyBiscuit) [inline extra-utils/premake/autobuild/build]: Use verbose. `-v`

File: extra-utils/premake/autobuild/build [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# build script copied from Arch Linux
+abinfo "Building premake ..."
+cd "build/gmake2.unix"
```
Comment (KexyBiscuit) [inline extra-utils/premake/autobuild/build]: Absolute path everywhere, like `cd "$SRCDIR"`

File: extra-utils/premake/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=premake
+PKGSEC=utils
+PKGDEP="glibc"
```
Comment (KexyBiscuit) [inline extra-utils/premake/autobuild/defines:3]: No need to list `glibc` here.

**Review by KexyBiscuit — CHANGES_REQUESTED** [review]

File: extra-utils/premake/autobuild/build (hunk 4 above)
Comment (MingcongBai) [reply extra-utils/premake/autobuild/build]: Also, use parenthetical expression.

```bash
(
    cd "$SRCDIR"/build/gmake2.unix
    ...
)

File: extra-utils/premake/autobuild/build [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+make STRIP="ture" config=release
+
+abinfo "Installing premake and LISENSE"
+cd ../..
+install -Dm755 "bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
+install -Dm644 LICENSE.txt "$PKGDIR/usr/share/licenses/$PKGNAME/LICENSE"
```
Comment (MingcongBai) [inline extra-utils/premake/autobuild/build]: Verbosity here, also absolute path, as with everywhere else.

File: extra-utils/premake/autobuild/build (hunk 3 above)
Comment (MingcongBai) [reply extra-utils/premake/autobuild/build]: Absolute path.

File: extra-utils/premake/spec [hunk 7]
```diff
@@ -0,0 +1,5 @@
+VER=5.0~a16
```
Comment (MingcongBai) [inline extra-utils/premake/spec]: Use 5.0.0~alpha16, and `${VER/\~/-}` in `SRCS`.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-utils/premake/autobuild/build [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# build script copied from Arch Linux
+abinfo "Building premake ..."
+cd "$PKGDIR/build/gmake2.unix"
```
Comment (MingcongBai) [inline extra-utils/premake/autobuild/build]: Make sure you understand what `$PKGDIR` and `$SRCDIR` are respetively.

File: extra-utils/premake/autobuild/build [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+make STRIP="true" config=release
+
+abinfo "Installing premake and LISENSE file"
+cd ../..
```
Comment (MingcongBai) [inline extra-utils/premake/autobuild/build]: Please just use parenthetical expression instead, cd-ing back is a waste of keyboard.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-utils/premake/autobuild/build [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# build script copied from Arch Linux
+abinfo "Building premake ..."
+cd "$SRCDIR/build/gmake2.unix"
+make STRIP="true" config=release
+
+abinfo "Installing premake and LISENSE file"
+cd "$SRCDIR"
+install -v -Dm755 "bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
```
Comment (MingcongBai) [inline extra-utils/premake/autobuild/build]: Use absolute path here.

File: extra-utils/premake/autobuild/build [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+install -v -Dm644 LICENSE.txt "$PKGDIR/usr/share/licenses/$PKGNAME/LICENSE"
```
Comment (MingcongBai) [inline extra-utils/premake/autobuild/build]: Use absolute path here.

File: extra-utils/premake/autobuild/build (hunk 10 above)
Comment (KexyBiscuit) [inline extra-utils/premake/autobuild/build]: Merge the options.
```suggestion
install -Dvm755 "bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
```

File: extra-utils/premake/autobuild/build (hunk 11 above)
Comment (KexyBiscuit) [inline extra-utils/premake/autobuild/build]:
```suggestion
install -Dvm644 LICENSE.txt "$PKGDIR/usr/share/licenses/$PKGNAME/LICENSE"
```

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-utils/premake/autobuild/build (hunk 10 above)
Comment (KexyBiscuit) [inline extra-utils/premake/autobuild/build]: Use absolute path everywhere.
```suggestion
install -v -Dm755 "$SRCDIR/bin/release/premake5" "${PKGDIR}/usr/bin/premake5"
```

File: extra-utils/premake/autobuild/build (hunk 11 above)
Comment (KexyBiscuit) [inline extra-utils/premake/autobuild/build]:
```suggestion
install -v -Dm644 "$SRCDIR"/LICENSE.txt "$PKGDIR/usr/share/licenses/$PKGNAME/LICENSE"
```

**Review by KexyBiscuit — DISMISSED** [review]

File: extra-utils/premake/autobuild/build [hunk 12]
```diff
@@ -0,0 +1,12 @@
+# build script adapted from Arch Linux
+abinfo "Building premake ..."
+(
+  cd "${SRCDIR}/build/gmake2.unix"
+  make STRIP="true" config=release
+)
+
+abinfo "Installing premake ..."
+(
+  cd "${SRCDIR}"
```
Comment (MingcongBai) [inline extra-utils/premake/autobuild/build]: This is not necessary, Autobuild3 runs the build scripts in "$SRCDIR".

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-utils/premake/autobuild/build [hunk 13]
```diff
@@ -0,0 +1,11 @@
+# build script adapted from Arch Linux
+abinfo "Building premake ..."
+(
+  cd "${SRCDIR}/build/gmake2.unix"
+  make STRIP="true" config=release
+)
+
+abinfo "Installing premake ..."
+(
```
Comment (MingcongBai) [inline extra-utils/premake/autobuild/build]: Why is this even necessary here?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-utils/premake/autobuild/build (hunk 13 above)
Comment (xyq1113723547) [reply extra-utils/premake/autobuild/build]: Isolating build and install set of commands is beneficial.
If there are errors in make stage, this info will not print out, which suggests errors are in pre-install stage.

File: extra-utils/premake/autobuild/defines [hunk 14]
```diff
@@ -0,0 +1,5 @@
...
+PKGDES="A simple build configuration tool for generating Makefiles"
+ABSPLITDBG=0
```
Comment (WhiredPlanck) [inline extra-utils/premake/autobuild/defines]: Please split debug info as much as possible, or explain why it's impossible to be done with a comment. See [`ABSPLITDBG` section](https://wiki.aosc.io/developer/packaging/autobuild3-manual/#:~:text=1%20or%20%E2%80%9Con%E2%80%9D\).-,absplitdbg,-expects%20a%20binary) below [Autobuild3 Manual - Clang specific configurations](https://wiki.aosc.io/developer/packaging/autobuild3-manual/#clang-specific-configurations).

**Review by WhiredPlanck — DISMISSED** [review]

**Comment by WhiredPlanck** [comment]
LGTM on AMD64

图片

**Comment by WhiredPlanck** [comment]
> LGTM on AMD64
> 
> 图片

This test process and results are still valid, only the version number has changed.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**